lychee-bin/tests/cli.rs

```diff
@@ -606,6 +606,80 @@ mod cli {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn test_lycheecache_file() -> Result<()> {
+        let mut cmd = main_command();
+
+        let test_base_path = fixtures_path().join("cache");
+        assert!(
+            fs::metadata(test_base_path.join(".lycheecache")).is_err(),
+            ".lycheecache file should not exist before this test"
+        );
+
+        let mock_server_ok = mock_server!(StatusCode::OK);
+        let mock_server_err = mock_server!(StatusCode::NOT_FOUND);
+        let mock_server_exclude = mock_server!(StatusCode::OK);
+
+        let dir = tempfile::tempdir()?;
+        let mut file = File::create(dir.path().join("c.md"))?;
+
+        writeln!(file, "{}", mock_server_ok.uri().as_str())?;
+        writeln!(file, "{}", mock_server_err.uri().as_str())?;
+        writeln!(file, "{}", mock_server_exclude.uri().as_str())?;
+
+        // run first without cache to generate the .lycheecache file
+        cmd.current_dir(&test_base_path)
+            .arg(dir.path().join("c.md"))
+            .arg("--verbose")
+            .arg("--cache")
+            .arg("--exclude")
+            .arg(mock_server_exclude.uri())
+            .assert()
+            .stderr(contains(format!("[200] {}/\n", mock_server_ok.uri())))
+            .stderr(contains(format!(
+                "[404] {}/ | Network error: Not Found\n",
+                mock_server_err.uri()
+            )))
+            .stderr(contains(format!(
+                "[EXCLUDED] {}/ | Excluded\n",
+                mock_server_exclude.uri()
+            )));
+
+        // check content of .lycheecache file
+        let data = fs::read_to_string(fixtures_path().join("cache").join(".lycheecache"))?;
+        assert!(data.contains(&format!("{}/,200", mock_server_ok.uri())));
+        assert!(data.contains(&format!("{}/,404", mock_server_err.uri())));
+        assert!(data.contains(&format!("{}/,Excluded", mock_server_exclude.uri())));
+
+        let mut cmd = main_command();
+
+        // run again to verify cache behavior
+        cmd.current_dir(&test_base_path)
+            .arg(dir.path().join("c.md"))
+            .arg("--verbose")
+            .arg("--cache")
+            .arg("--exclude")
+            .arg(mock_server_exclude.uri())
+            .assert()
+            .stderr(contains(format!(
+                "[200] {}/ | OK (cached)\n",
+                mock_server_ok.uri()
+            )))
+            .stderr(contains(format!(
+                "[404] {}/ | Error (cached)\n",
+                mock_server_err.uri()
+            )))
+            .stderr(contains(format!(
+                "[EXCLUDED] {}/ | Excluded\n",
+                mock_server_exclude.uri()
+            )));
+
+        // clear the .lycheecache file
+        fs::remove_file(fixtures_path().join("cache").join(".lycheecache"))?;
```

Comment:
Same here, `fixtures_path().join("cache")` could be replaced with `test_base_path().join(".lycheecache")` for consistency. 
Alternatively you could also introduce a `cache_file` variable in the beginning and reuse that throughout the test. 😉 
```rust
let cache_file = test_base_path().join(".lycheecache");
```

Reply:
Good idea! I'll make that update